etcdserver/apply.go

```diff
@@ -53,7 +53,7 @@ type applierV3 interface {
 	Put(txnID int64, p *pb.PutRequest) (*pb.PutResponse, error)
 	Range(txnID int64, r *pb.RangeRequest) (*pb.RangeResponse, error)
 	DeleteRange(txnID int64, dr *pb.DeleteRangeRequest) (*pb.DeleteRangeResponse, error)
-	Txn(rt *pb.TxnRequest) (*pb.TxnResponse, error)
+	Txn(header *pb.RequestHeader, rt *pb.TxnRequest) (*pb.TxnResponse, error)
```

Comment:
+1. Having this auth check in Txn seems inconsistent to me, looking at other methods in the same interface